readthedocs/search/management/commands/reindex_elasticsearch.py

```diff
@@ -19,17 +18,27 @@ class Command(BaseCommand):
 
     @staticmethod
     def _get_indexing_tasks(app_label, model_name, index_name, queryset, document_class):
-        total = queryset.count()
-        chunks = get_chunk(total, settings.ES_TASK_CHUNK_SIZE)
-
-        for chunk in chunks:
-            data = {
-                'app_label': app_label,
-                'model_name': model_name,
-                'document_class': document_class,
-                'index_name': index_name,
-                'chunk': chunk
-            }
+        chunk_size = settings.ES_TASK_CHUNK_SIZE
+        qs_iterator = queryset.only('pk').iterator()
+        is_iterator_empty = False
+
+        data = {
+            'app_label': app_label,
+            'model_name': model_name,
+            'document_class': document_class,
+            'index_name': index_name,
+        }
+
+        while not is_iterator_empty:
```

Comment:
Actually you can just use `islice` from `itertools`

```python
from itertools import islice

objects_id = list(islice(qs_iterator, chunk_size))
```

Reply:
And you can use https://docs.djangoproject.com/en/2.2/ref/models/querysets/#values-list to only get an iterator with only the ids

Reply:
With this approach we will have a list of all the ids at once.
I think it is better to iterate over on generator object?

Reply:
It returns a `QuerySet` of values, it's lazy https://stackoverflow.com/questions/37140426/does-django-queryset-values-list-return-a-list-object

Reply:
Ohhh...
But we also want to logging based on the pk.
Sorry... But I'm not able to think on how to use this method and to log also.
Do you have something on your mind?

Reply:
you can get the last pk from the list `last_pk = objects_id[-1] if objects_id else 0`

Reply:
Not so important anyway, current logic works too

Reply:
I prefer the current implementation. It feels more explicit to me.